violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/EditorPart.java

```diff
@@ -181,6 +214,72 @@ public void removeSelected()
         }
     }
 
+    /*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.horstmann.violet.framework.display.clipboard.IEditorPart#
+	 * convertSelected()
+	 */
+	public void convertSelected() {
+		String[] options = new String[] {CONVERT_WARN_YES, CONVERT_WARN_NO};
+	    int response = JOptionPane.showOptionDialog(null, CONVERT_WARN_TEXT, CONVERT_WARN_TITLE,
+	        JOptionPane.DEFAULT_OPTION, JOptionPane.WARNING_MESSAGE,
+	        null, options, options[0]);
+	    if(response!=0){
+	    	return;
+	    }
+	    
+	    List<INode> selectedNodes = selectionHandler.getSelectedNodes();
+		this.behaviorManager.fireBeforeRemovingSelectedElements();
+		
+		for (INode node : selectedNodes) {
+			if (node instanceof ISwitchableNode) {
+				ISwitchableNode switchableNode = (ISwitchableNode) node;
+				INode convertedNode = switchableNode.switchNode();
+				if (convertedNode != null) {
+					graph.removeNode(node);
+					addNodeAtPoint(convertedNode, node.getLocationOnGraph());
+				} else {
+					JOptionPane.showMessageDialog(this,
+							CONVERT_ERROR_TEXT,
+							CONVERT_ERROR_TITLE,
+						    JOptionPane.ERROR_MESSAGE);
+					System.exit(-1);
```

Comment:
But i got instruction to catch it and add app terminate - so i did it. Before it was just continue in this loop, so nothing great will happened.